types/filters/parse.go

```diff
@@ -8,6 +8,8 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+
+	versionpkg "github.com/docker/docker/pkg/version"
```

Comment:
you can just copy this function and use it to compare the two strings directly:

https://github.com/docker/docker/blob/master/pkg/version/version.go#L11-L38